neuraxle/data_container.py

```diff
@@ -136,12 +157,62 @@ def convolved_1d(self, stride, kernel_size) -> Iterable['DataContainer']:
                     expected_outputs = expected_outputs[:i]
                     break
 
-            yield DataContainer(data_inputs=data_inputs, current_ids=current_ids, summary_id=self.summary_id,
-                                expected_outputs=expected_outputs)
+            yield DataContainer(
+                data_inputs=data_inputs,
+                current_ids=current_ids,
+                summary_id=self.summary_id,
+                expected_outputs=expected_outputs,
+                sub_data_containers=self.sub_data_containers
+            )
 
     def copy(self):
-        return DataContainer(data_inputs=self.data_inputs, current_ids=self.current_ids, summary_id=self.summary_id,
-                             expected_outputs=self.expected_outputs)
+        return DataContainer(
+            data_inputs=self.data_inputs,
+            current_ids=self.current_ids,
+            summary_id=self.summary_id,
+            expected_outputs=self.expected_outputs,
+            sub_data_containers=[(name, data_container.copy()) for name, data_container in self.sub_data_containers]
```

Comment:
Shouldn't `sub_data_containers` be a dict rather than a tuple list?

Reply:
can wee fix this in  a futur epr https://github.com/Neuraxio/Neuraxle/issues/175